OutOfSchool/OutOfSchool.WebApi/Services/Images/ImageDependentEntityImagesInteractionMediator.cs

```diff
@@ -21,152 +21,168 @@ namespace OutOfSchool.WebApi.Services.Images
+        where TEntity : class, IKeyedEntity, IImageDependentEntity<TEntity>, new()
     {
         /// <summary>
-        /// Initializes a new instance of the <see cref="ImageInteractionBaseService{TEntity, TKey}"/> class.
+        /// Initializes a new instance of the <see cref="ImageDependentEntityImagesInteractionMediator{TEntity}"/> class.
         /// </summary>
         /// <param name="imageService">Service for interacting with an image storage.</param>
-        /// <param name="repository">Repository with images.</param>
         /// <param name="limits">Describes limits of images for entities.</param>
         /// <param name="logger">Logger.</param>
-        public ImageInteractionBaseService(
+        public ImageDependentEntityImagesInteractionMediator(
             IImageService imageService,
-            IEntityRepositoryBase<TKey, TEntity> repository,
-            ImagesLimits<TEntity> limits,
-            ILogger<ImageInteractionBaseService<TEntity, TKey>> logger)
+            IOptions<ImagesLimits<TEntity>> limits,
+            ILogger<ImageDependentEntityImagesInteractionMediator<TEntity>> logger)
         {
             ImageService = imageService;
-            Repository = repository;
-            Limits = limits;
+            Limits = limits.Value;
             Logger = logger;
         }
 
         /// <summary>
         /// Gets a logger.
         /// </summary>
-        protected ILogger<ImageInteractionBaseService<TEntity, TKey>> Logger { get; }
+        protected ILogger<ImageDependentEntityImagesInteractionMediator<TEntity>> Logger { get; }
 
         /// <summary>
         /// Gets a service for interacting with an image storage..
         /// </summary>
         protected IImageService ImageService { get; }
 
-        /// <summary>
-        /// Gets a repository with images.
-        /// </summary>
-        protected IEntityRepositoryBase<TKey, TEntity> Repository { get; }
-
         /// <summary>
         /// Gets limits of images for entity of <c>TEntity</c> type.
         /// </summary>
         protected ImagesLimits<TEntity> Limits { get; }
 
         /// <inheritdoc/>
-        public async Task<Result<string>> UploadImageAsync(TKey entityId, IFormFile image)
+        public async Task<Result<string>> AddImageAsync(TEntity entity, IFormFile image)
+            => await UploadImageProcessAsync(entity, image).ConfigureAwait(false);
+
+        /// <inheritdoc/>
+        public async Task<OperationResult> RemoveImageAsync(TEntity entity, string imageId)
+            => await RemoveImageProcessAsync(entity, imageId).ConfigureAwait(false);
+
+        public async Task<ImageChangingResult> ChangeImageAsync(TEntity entity, string oldImageId, IFormFile newImage)
         {
-            if (image == null)
+            _ = entity ?? throw new ArgumentNullException(nameof(entity));
+            Logger.LogTrace("Updating an image the for entity was started");
```

Comment:
"Updating an image the for entity" - wording issue, probably excessive `the` article or missed noun